eventsource/repository.go

```diff
@@ -184,28 +189,37 @@ func (repo repository) Load(ctx context.Context, aggregateID string, aggr Aggreg
 	return
 }
 
-func (repo repository) GetEventsBySequenceID(ctx context.Context, sequenceID string) (events []Event, err error) {
-	var records []Record
-	records, err = repo.store.LoadBySequenceID(ctx, sequenceID)
+func unmarshalRecords(serializer Serializer, records []Record) (events []Event, err error) {
 	for _, record := range records {
 		var event Event
-		if event, err = repo.serializer.Unmarshal(record.Data, record.Type); err != nil {
+		if event, err = serializer.Unmarshal(record.Data, record.Type); err != nil {
 			return
 		}
 		events = append(events, event)
 	}
 	return
 }
 
+func (repo repository) GetEventsBySequenceID(ctx context.Context, sequenceID string) (events []Event, err error) {
+	var records []Record
+	if records, err = repo.store.LoadBySequenceID(ctx, sequenceID); err != nil {
+		return
+	}
+	return unmarshalRecords(repo.serializer, records)
+}
+
+func (repo repository) GetEventsBySequenceIDAndType(ctx context.Context, sequenceID string, eventType Event) (events []Event, err error) {
+	var records []Record
+	if records, err = repo.store.LoadBySequenceIDAndType(ctx, sequenceID, reflect.TypeOf(eventType).Name()); err != nil {
```

Comment:
`reflect.TypeOf(eventType).Name()` will not work if eventType is of type pointer

```
func getTypeOfValue(input interface{}) reflect.Type {
	value := reflect.TypeOf(input)
	if value.Kind() == reflect.Ptr {
		value = value.Elem()
	}
	return value
}
```

`getTypeOfValue(eventType).Name()` this will give you what you want, as long as the eventType isn't a double pointer :D

The function above already exists in the serilizer. Would be nice if you updated `SaveTransaction` that has the same fault

Reply:
Good catch. We could even handle double pointers if instead of `value = value.Elem()` we do `return getTypeOfValue(value.Elem())` :)